### ISSUE TITLE
registerMarker/registerDecoration in overviewRuler does not work if invoked immediately

### ISSUE THREAD

**Issue by jtbandes** [xtermjs/xterm.js#5214]
When writing a large batch of lines after clearing the terminal, and using `registerMarker`+`registerDecoration` to render lines in the overview ruler, the decorations do not appear if all the lines are written at once. If a small delay is introduced with `setTimeout`, the decorations work properly.

Additionally, a small decoration sometimes remains at the very top of the overview ruler even after the terminal is cleared.

https://github.com/user-attachments/assets/b52d0096-d6b7-455e-a082-d7ce251e3b69

## Details
- Browser and browser version: Chrome 130.0.6723.117
- OS version: macOS 15.1
- xterm.js version: 5.5.0

### Steps to reproduce

https://codesandbox.io/p/sandbox/xtermjs-test-forked-yxrd7j

**Comment by jtbandes** [comment]
It seems like the marker always has `line: 0`. Maybe this is because the newly written input has not yet been parsed?

Waiting for the new input to be parsed before adding the marker might introduce a lot of delay. Is there any way to do this without waiting, or to force processing of the write buffer?

**Comment by jtbandes** [comment]
Thank you! I think that works. It was not clear to me from docs that this callback is actually associated with the single line/chunk being written... I assumed it might be batched with other callbacks if the terminal was processing data in batches.

**Comment by jerch** [comment]
> It was not clear to me from docs that this callback is actually associated with the single line/chunk being written... I assumed it might be batched with other callbacks if the terminal was processing data in batches.

You are right - it is not associated with single line writing, it is just a callback denoting "this chunk of data is fully parsed and processed by the terminal buffer". To get "line-wise semantics" here, you'd have to chunk the data into lines upfront, and would get a callback for each line this way.

**Comment by jtbandes** [comment]
But just to confirm, if I call write twice in a row and pass two callbacks, whenever the data is processed the callbacks will not be batched together but will immediately invoked after the corresponding data chunk is processed? e.g.
- process chunk 1
- call callback 1
- process chunk 2
- call callback 2

even if this processing is happening all together?
This is what was not clear to me from the docs :)

**Comment by jerch** [comment]
Yes the callback gets triggered synchronous right after the chunk got processed. So no new chunk data will be handled yet.
